server/plugin.go

```diff
@@ -45,8 +52,35 @@ type Plugin struct {
 
 func (p *Plugin) OnActivate() error {
 	p.logger = &metricsLogger{api: p.API}
-
 	appCfg := p.API.GetConfig()
+
+	p.closeChan = make(chan bool)
+	p.waitGroup = sync.WaitGroup{}
+
+	// we are using a mutually exclusive lock to run a single instance of this plugin
+	// we don't really need to collect metrics twice: although TSDB will take care
+	// of overlapped blocks, it will increase the disk writes to the remote or local
+	// disk.
+	if p.isHA() {
+		var err error
+		p.singletonLock, err = cluster.NewMutex(p.API, "metrics")
```

Comment:
Should we use the pluginID as the key to avoid clashes in case the customer has a custom plugin which uses the same mutex key name?